### PR TITLE
Ocean Waves: harmonize initial profile

### PR DESCRIPTION
**Pull request by mbkuhn**
## Summary

This sets up the boundary regions better at initialization and smooths the profile to whatever is specified within the bulk of the domain. Did some refactoring along the way. This PR helps with some stability issues we've been seeing by avoiding initial transients that were difficult for the pressure solver.

## Pull request type

not exactly sure:

- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Checklist

The following is included:

- [x] new unit-test(s)
- [ ] new regression test(s)
- [ ] documentation for new capability

This PR was tested by running:

- the unit tests
  - [ ] on GPU 
  - [x] on CPU 
- the regression tests
  - [ ] on GPU 
  - [ ] on CPU 

## Additional background

needs new unit tests

### CODE REVIEW

**Comment by mbkuhn** [comment]
unit tests added. This will cause diffs in the ow_linear* and ow_stokes reg tests. I need to make similar changes (for the sake of robustness) to the waves2amr pathway, but I'll save that for a future PR.

**Review by mpolimeno — APPROVED** [review]
Really cool. Thanks for doing this @mbkuhn !

File: amr-wind/ocean_waves/relaxation_zones/linear_waves_ops.H [hunk 1]
```diff
@@ -43,69 +43,72 @@ struct InitDataOp<LinearWaves>
...
+
+                    // Quiescent profile
+                    const utils::WaveVec quiescent{
+                        0.0, 0.0, 0.0, zero_sea_level};
+
+                    // Specify initial state for each region of domain
+                    const auto bulk = init_wave_field ? wave_sol : quiescent;
+                    const auto outlet = has_beach ? quiescent : wave_sol;
+
+                    auto local_profile = utils::harmonize_profiles_1d(
```
Comment (moprak-nrel) [inline amr-wind/ocean_waves/relaxation_zones/linear_waves_ops.H:99]: const this?

File: amr-wind/ocean_waves/relaxation_zones/stokes_waves_ops.H [hunk 2]
```diff
@@ -92,46 +93,47 @@ struct InitDataOp<StokesWaves>
...
+
+                    // Quiescent profile
+                    const utils::WaveVec quiescent{
+                        0.0, 0.0, 0.0, zero_sea_level};
+
+                    // Specify initial state for each region of domain
+                    const auto bulk = init_wave_field ? wave_sol : quiescent;
+                    const auto outlet = has_beach ? quiescent : wave_sol;
+
+                    auto local_profile = utils::harmonize_profiles_1d(
```
Comment (moprak-nrel) [inline amr-wind/ocean_waves/relaxation_zones/stokes_waves_ops.H:124]: const?

**Review by moprak-nrel — COMMENTED** [review]
Looks good! couple of super minor comments

File: amr-wind/ocean_waves/relaxation_zones/linear_waves_ops.H (hunk 1 above)
Comment (mbkuhn) [reply amr-wind/ocean_waves/relaxation_zones/linear_waves_ops.H:99]:
```suggestion
                    const auto local_profile = utils::harmonize_profiles_1d(
```

File: amr-wind/ocean_waves/relaxation_zones/stokes_waves_ops.H (hunk 2 above)
Comment (mbkuhn) [reply amr-wind/ocean_waves/relaxation_zones/stokes_waves_ops.H:124]:
```suggestion
                    const auto local_profile = utils::harmonize_profiles_1d(
```

**Review by moprak-nrel — APPROVED** [review]

**Pull request merged**